demisto_sdk/commands/test_content/TestContentClasses.py

```diff
@@ -3244,3 +3251,99 @@ def __str__(self):
 
     def __repr__(self):
         return str(self)
+
+
+class ContentStatusUpdater:
+    def __init__(self, artifacts_folder: Path) -> None:
+        """
+        Initializes the ContentStatusUpdater with the folder and content status filename.
+
+        Args:
+            artifacts_folder (str): The folder where the content status file is located.
+        """
+        self.artifacts_folder = artifacts_folder
+        self.content_status_filename = "content_status.json"
+        self.content_status_path = Path(artifacts_folder) / self.content_status_filename
+        self.content_status: Dict[Any, Any] = {}
+
+    def update_content_status(
+        self, successful_tests: List[str], failed_tests: List[str]
+    ) -> None:
+        """
+        Updates the content status with the provided test results, adding the failed and successful playbooks.
+
+        Args:
+            successful_tests (List[str]): List of successful playbooks to be added.
+            failed_tests (List[str]): List of failed playbooks to be added.
+        """
+        logging.info(
+            f"Starting update_content_status with {len(failed_tests)} failed tests and "
+            f"{len(successful_tests)} successful tests."
+        )
+
+        self._load_content_status()
+        self._initialize_content_status_keys()
+
+        self._update_playbooks("failed_playbooks", failed_tests)
+        self._update_playbooks("successful_playbooks", successful_tests)
+
+        self._save_content_status()
+
+    def _load_content_status(self) -> None:
+        """
+        Attempts to load the content status from the file. If the file doesn't exist or is invalid, initializes an empty status.
+        """
+        if Path.exists(self.content_status_path):
+            logging.info(f"Content status file exists at {self.content_status_path}")
+            try:
+                with open(self.content_status_path, "r") as content_file:
+                    self.content_status = json.load(content_file)
+                    logging.info(f"Loaded content status: {self.content_status}")
+            except json.JSONDecodeError as e:
+                logging.error(
+                    f"JSON decode error while loading content_status.json: {e}"
+                )
```

Comment:
Shouldn't we exit if there's an invalid json?

Reply:
There shouldn't be, but since it's running in the build my thought was to log it and proceed with a new one on the off chance it wasn't created. Typically in the configure and install packs set, we either populate it with info there or just create an empty one. Later in the build we aggregate every status file so it's less of a concern.